intervals/main_objs.py

```diff
@@ -1576,9 +1706,55 @@ def _split_by_threshold(seq, max_diff=70):
 
         yield part
 
+    # July 2022 added to check for overlap
+    def _dur_ngram_helper(df, ng_durs):
+        """
+        This helper is used by the presentation type classifier to check
+        total duration of each ngram in a given set of soggetti for a
+        presentation type.
+
+        Note that it relies on the ng_dur dataframe that appears in the
+        presentation type classifier itself:
+
+        ng_durs = self.durations(df=entries)
+
+        These in turn are saved as a string to check for overlaps among entries.
+        """
+        dur_list = []
+        l = list(zip(df["Offsets"], df["Voices"]))
+        for pair in l:
+            dur_of_ng = ng_durs.loc[pair[0], pair[1]]
+            dur_list.append(dur_of_ng)
+        return dur_list
+
+    # July 2022 added to check for overlap
+    def _entry_overlap_helper(df):
+        """
+        This private function is used as part of the last stage of the Presentation
+        Type classifier.  It reports the overlap (in offsets) between successive entries
+        in a given point of imitation.
+        """
+        endpoints = [x + y for x, y in zip(df["Entry_Durs"], df["Offsets"])]
```

Comment:
You can also do math directly on dfs and series, as in:
endpoints = df["Entry_Durs"] + df["Offsets"]

Reply:
not sure, those are lists, I think.  That's why we need the Zip.  But perhaps I am missing something